index.js

```diff
@@ -24,8 +47,13 @@ module.exports = class Emittery {
 
 	on(eventName, listener) {
 		assertEventName(eventName);
-		this._getListeners(eventName).add(listener);
-		return this.off.bind(this, eventName, listener);
+
+		if (typeof listener === 'function') {
```

Comment:
Should this check the number of arguments passed, rather than the `listener` type? It seems a little too easy to accidentally create an asynchronous iterator otherwise.

Reply:
As you wish. Although it is a public method with a specific contract so I think this alternative is better. Otherwise it means I'll need to check for "arguments" which is not correct in strict mode; Or use the spread operator on the signature which makes the contract less obvious in my opinion